Remove flashmessenger view helper

gsteel opened the pull request:
### Description

Rebases and resolves #47 

Also closes #8 
Closes #16 

As this is the last remnant of prophecy usage, this pull also removes `phpspec/prophecy` from dev deps

Closes #133

samsonasik committed:
Fixes #16 remove FlashMessenger view helper

Signed-off-by: Abdul Malik Ikhsan <samsonasik@gmail.com>

gsteel committed:
- Removes flash messenger from docs build
- Removes magic method docs for removed helpers
- Removes stray factory definition for removed helper

Signed-off-by: George Steel <george@net-glue.co.uk>

gsteel committed:
Updates baseline after removal of flash messenger

Signed-off-by: George Steel <george@net-glue.co.uk>

Ocramius reviewed (approved):
Nice, didn't even know it was split out!

gsteel commented:
7.4 on lowest looks stuck - can you re-trigger please @Ocramius

gsteel committed:
Lock update post rebase.
Remove escaper from dev deps - it crept back in during a merge and is already present in deps
Remove flash messenger from dev deps. It is no longer referenced anywhere.

Signed-off-by: George Steel <george@net-glue.co.uk>

Ocramius commented:
GHA didn't start at all here :thinking:

Ocramius commented:
Welp, it's broken again: https://www.githubstatus.com/

gsteel commented:
Not again 🙄

gsteel committed:
Remove phpspec/prophecy from dev dependencies

Closes #133

Tests removed in this branch contain the last remaining usage of Prophecy

Signed-off-by: George Steel <george@net-glue.co.uk>

Ocramius commented:
I think pushing an empty commit would give us feedback here - really sorry to push it on contributors, but can't do it from here ATM :|

gsteel commented:
> I think pushing an empty commit would give us feedback here - really sorry to push it on contributors, but can't do it from here ATM :|

I think GHA is still down isn't it?

Ocramius commented:
Seems to be working/picking up stuff - the hook system was completely broken though

gsteel commented:
Appears to have gone green despite the UI not showing the checks: https://github.com/laminas/laminas-view/actions/runs/2029729860

Ocramius commented on composer.json:
```diff
@@ -41,22 +41,18 @@
     "require-dev": {
         "laminas/laminas-authentication": "^2.5",
         "laminas/laminas-coding-standard": "~2.3.0",
-        "laminas/laminas-escaper": "^2.5",
```
Is escaper supposed to be gone?

gsteel replied:
It's not gone, it was moved to `require` in 2.x but a merge popped it back in `require-dev` too, so I removed it from `require-dev` in this pull to clean up. I must have missed this resolving conflicts earlier today.

Ocramius reviewed (approved):
LGTM 🚢

Ocramius commented:
Thanks @gsteel!
